### PR TITLE
Auto-generate source schema upon __new__ `BaseDataInterface`

### PR DESCRIPTION
**Pull request by weiglszonja**
Opened as a follow-up on https://github.com/catalystneuro/neuroconv/pull/372

### CODE REVIEW

File: src/neuroconv/utils/json_schema.py [hunk 1]
```diff
@@ -63,9 +63,9 @@ def get_schema_from_method_signature(method: Callable, exclude: list = None) ->
 
     """
     if exclude is None:
-        exclude = ["self", "kwargs"]
+        exclude = ["cls", "self", "kwargs"]
     else:
-        exclude = exclude + ["self", "kwargs"]
+        exclude = exclude + ["cls", "self", "kwargs"]
```
Comment (weiglszonja) [inline src/neuroconv/utils/json_schema.py:68]: @bendichter thoughts on this? without adding to exclude I got this: 

```
  File "/Users/weian/Library/Mobile Documents/com~apple~CloudDocs/catalystneuro/neuroconv/src/neuroconv/utils/json_schema.py", line 119, in get_schema_from_method_signature
    raise ValueError(
ValueError: No valid arguments were found in the json type mapping '<class 'inspect._empty'>' for parameter cls
```

File: src/neuroconv/basedatainterface.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ class BaseDataInterface(ABC):
     @classmethod
     def get_source_schema(cls):
         """Infer the JSON schema for the source_data from the method signature (annotation typing)."""
-        return get_schema_from_method_signature(cls.__init__, exclude=["source_data"])
+        return get_schema_from_method_signature(cls.__new__, exclude=["source_data"])
```
Comment (weiglszonja) [inline src/neuroconv/basedatainterface.py:17]: the tests are failing so something else is missing here, any suggestions would be appreciated, not sure how to fix.

**Comment by bendichter** [comment]
@weiglszonja does this work for you?

**Comment by weiglszonja** [comment]
@bendichter indeed, thanks!

**Review by bendichter — APPROVED** [review]

**Pull request merged**